android-webrtc/URLVideoCapturer/urlcapturer/src/main/java/com/amazonaws/kinesisvideo/webrtc/urlvideocapturer/URLVideoCapturer.java

```diff
@@ -0,0 +1,140 @@
+package com.amazonaws.kinesisvideo.webrtc.urlvideocapturer;
+
+import android.content.Context;
+import android.net.Uri;
+
+import org.videolan.libvlc.LibVLC;
+import org.videolan.libvlc.Media;
+import org.videolan.libvlc.MediaPlayer;
+import org.videolan.libvlc.interfaces.IVLCVout;
+import org.webrtc.CapturerObserver;
+import org.webrtc.SurfaceTextureHelper;
+import org.webrtc.VideoCapturer;
+import org.webrtc.VideoFrame;
+import org.webrtc.VideoSink;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * This class implements a WebRTC VideoCapturer for media streams. Similar to how camera capturers work, this
+ * class also leverages the SurfaceTextureHelper to do the heavy lifting of grabbing and encoding video
+ * frames using an intermediate OpenGL texture. It also uses libVLC(User must add) to playback streams on the texture.
+ */
+public class URLVideoCapturer implements VideoCapturer, VideoSink {
+    private String url;
+    private String[] options;
+    private String aspectRatio;
+    private SurfaceTextureHelper surfaceTextureHelper;
+    private Context context;
+    private CapturerObserver capturerObserver;
+
+    /**
+     * Public constructor accepting the stream url and libVLC configuration.
+     *
+     * @param url         Media stream url.
+     * @param options     List of options to pass on to libVLC.
+     * @param aspectRatio Aspect ratio to configure for libVLC.
+     */
+    public URLVideoCapturer(String url, String[] options, String aspectRatio) {
+        this.url = url;
```

Comment:
Did we want to perform validation of the media url (regex)? Or maybe require a Uri object be passed instead of a String?

Reply:
Updated the code to accept an android.net.Uri object instead of String.